src/bio.c

```diff
@@ -74,19 +74,23 @@ static list *bio_jobs[BIO_NUM_OPS];
  * the sensible operation. This data is also useful for reporting. */
 static unsigned long long bio_pending[BIO_NUM_OPS];
 
+struct lazy_free_job {
+    lazy_free_fn *fn; /* Function that will free the provided arguments */
+    void *args[1]; /* List of arguments to be passed to the free function */
```

Comment:
It would be cleaner, but it throws a warning:
"warning: zero size arrays are an extension [-Wzero-length-array]". I suppose it's more portable this way. I think it's okay because we need at least one argument.

Reply:
i you can use an empty array then, like we use here:
```c
typedef struct clientReplyBlock {
    size_t size, used;
    char buf[];
} clientReplyBlock;
```
that seems the portable way to do it (C99).

Reply:
TIL :)

Reply:
For current bio use, it's ok, but i think it should be more generic, what should we do if we we add a new bio job, such as dump and restore keys.